Samples/AudioPreprocessing/WinUI/AudioPreprocessing/AudioPreprocessing/Model/PreprocessingModel.cs

```diff
@@ -0,0 +1,259 @@
+﻿using NAudio.Wave;
+using System;
+using System.Collections.Generic;
+using Microsoft.AI.MachineLearning.Experimental;
+using Microsoft.AI.MachineLearning;
+using Windows.Media;
+using Windows.Graphics.Imaging;
+
+using Operator = Microsoft.AI.MachineLearning.Experimental.LearningModelOperator;
+using System.Diagnostics;
+
+namespace AudioPreprocessing.Model
+{
+    public class PreprocessModel
+    {
+        const string MS_EXPERIMENTAL_DOMAIN = "com.microsoft.experimental";
+
+        public string AudioPath { get; set; }
+
+        public string MelSpecImagePath { get; set; }
+
+        public PreprocessModel()
+        {
+            AudioPath = "..\\tmp\\mel_spectrogram_file.jpg";
+            MelSpecImagePath = "..\\tmp\\mel_spectrogram_file.jpg";
+        }
+
+        public SoftwareBitmap GenerateMelSpectrogram(string audioPath)
+        {
+            var signal = GetSignalFromFile(audioPath);
+            int batchSize = 1;
+            int signalSize = signal.Count;
+            int windowSize = 256;
+            int dftSize = 256;
+            int hopSize = 3;
+            int nMelBins = 1024;
+            int samplingRate = 8192;
+            var rawSoftwareBitmap = GetMelspectrogramFromSignal(signal, batchSize, signalSize, windowSize, dftSize,
+                hopSize, nMelBins, samplingRate);
+
+            return rawSoftwareBitmap;
+        }
+
+        private List<float> Resample(List<float> signal, int oldSampleRate, int newSampleRate)
+        {
+            int nStep = oldSampleRate / newSampleRate;
+            var result = new List<float>();
```

Comment:
Dont make this a List<float>.

The resampled signal size is constant sized, and has frequent element access. An Array would be better.

var result = Array.CreateInstance(typeof(float), signal.Count);

When you return the Array be sure to cast it to IEnumerable<float>. This works, but the conversion is not implicit.

return result as IEnumerable<float>;

Reply:
Resample was just a temporary debugging tool from a while ago, so I've removed it, but GetSignalFromFile uses arrays now.